src/execution_tree/primitives/phyname.cpp

```diff
@@ -0,0 +1,108 @@
+
+///////////////////////////////////////////////////////////////////////////////
+namespace phylanx { namespace execution_tree { namespace primitives
+{
+    ///////////////////////////////////////////////////////////////////////////
+    primitive create_phyname(hpx::id_type const& locality,
+        primitive_arguments_type&& operands,
+        std::string const& name, std::string const& codename)
+    {
+        static std::string type("phyname");
+        return create_primitive_component(
+            locality, type, std::move(operands), name, codename);
+    }
+
+    match_pattern_type const phyname::match_data =
+    {
+        hpx::util::make_tuple("phyname",
+            std::vector<std::string>{"phyname(_1)"},
+            &create_phyname, &create_primitive<phyname>,
+            R"(arg
+            Args:
+
+                arg object : return the phylanx type name for the object
+
+            Returns:)"
+            )
+    };
+
+    ///////////////////////////////////////////////////////////////////////////
+    phyname::phyname(
+            primitive_arguments_type&& operands,
+            std::string const& name, std::string const& codename)
+      : primitive_component_base(std::move(operands), name, codename)
+    {}
+
+    hpx::future<primitive_argument_type> phyname::eval(
+        primitive_arguments_type const& operands,
+        primitive_arguments_type const& args, eval_context ctx) const
+    {
+        auto this_ = this->shared_from_this();
+        return hpx::dataflow(hpx::launch::sync, hpx::util::unwrapping(
+            [this_ = std::move(this_)](primitive_arguments_type&& args)
+                -> primitive_argument_type
+            {
+                if(args.size() == 0) {
+                    std::string name = "None";
+                    return primitive_argument_type{name};
+                }
+                std::string name;
+                int tid = args[0].index();
+                switch(tid) {
+                case 0:
+                    name = "None";
+                    break;
+                case 1:
+                    name = "node_data<uint8>";
+                    break;
+                case 2:
+                    name = "node_data<int64>";
+                    break;
+                case 3:
+                    name = "string";
+                    break;
+                case 4:
+                    name = "node_data<float>";
+                    break;
+                case 5:
+                    name = "primitive";
+                    break;
+                case 6:
+                    name = "vector<expression>";
+                    break;
+                case 7:
+                    name = "range";
+                    break;
+                case 8:
+                    name = "dict";
+                    break;
+                default:
+                    name  = "??";
+                    break;
+                }
+                return primitive_argument_type{name};
```

Comment:
This functionality is already available from the main phylanx module (see https://github.com/STEllAR-GROUP/phylanx/blob/master/src/execution_tree/primitives/base_primitive.cpp#L65). It just needs to be exposed to the world (as its currently a static/internal function).